src/Components/Components/src/Routing/Router.cs

```diff
@@ -114,8 +125,33 @@ private static string StringUntilAny(string str, char[] chars)
                 : str.Substring(0, firstIndex);
         }
 
+        private void RefreshRouteTable()
+        {
+            var assemblies = AdditionalAssemblies == null ? new[] { AppAssembly } : new[] { AppAssembly }.Concat(AdditionalAssemblies);
```

Comment:
Could we short-circuit this if `AdditionalAssemblies` is null which for most users is the default case? Or is there a scenario where the app assembly can be changed between refreshes?

Reply:
We do short-circuit refreshing the route table if additional assemblies is unchanged because we do the hashset comparison in the code below.

Note: the code currently uses a length check but it's replaced with an equality check in an upcoming commit.